tests/bsim/bluetooth/host/att/sequential/dut/src/main.c

```diff
@@ -0,0 +1,368 @@
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (conn_err) {
+		FAIL("Failed to connect to %s (%u)", addr, conn_err);
+		return;
+	}
+
+	LOG_DBG("%s", addr);
+
+	dconn = bt_conn_ref(conn);
+	SET_FLAG(is_connected);
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_DBG("%p %s (reason 0x%02x)", conn, addr, reason);
+
+	bt_conn_unref(dconn);
+	UNSET_FLAG(is_connected);
+}
+
+static void data_len_updated(struct bt_conn *conn,
+			     struct bt_conn_le_data_len_info *info)
+{
+	LOG_DBG("Data length updated: TX %d RX %d",
+		info->tx_max_len,
+		info->rx_max_len);
+	SET_FLAG(flag_data_length_updated);
+}
+
+static void do_dlu(void)
+{
+	int err;
+	struct bt_conn_le_data_len_param param;
+
+	param.tx_max_len = CONFIG_BT_CTLR_DATA_LENGTH_MAX;
+	param.tx_max_time = 2500;
+
+	err = bt_conn_le_data_len_update(dconn, &param);
+	ASSERT(err == 0, "Can't update data length (err %d)\n", err);
+
+	WAIT_FOR_FLAG(flag_data_length_updated);
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+	.le_data_len_updated = data_len_updated,
+};
+
+static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			 struct net_buf_simple *ad)
+{
+	struct bt_le_conn_param *param;
+	struct bt_conn *conn;
+	int err;
+
+	err = bt_le_scan_stop();
+	if (err) {
+		FAIL("Stop LE scan failed (err %d)", err);
+		return;
+	}
+
+	char str[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(addr, str, sizeof(str));
+
+	LOG_DBG("Connecting to %s", str);
+
+	param = BT_LE_CONN_PARAM_DEFAULT;
+	err = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, param, &conn);
+	if (err) {
+		FAIL("Create conn failed (err %d)", err);
+		return;
+	}
+}
+
+static void connect(void)
+{
+	struct bt_le_scan_param scan_param = {
+		.type = BT_LE_SCAN_TYPE_ACTIVE,
+		.options = BT_LE_SCAN_OPT_NONE,
+		.interval = BT_GAP_SCAN_FAST_INTERVAL,
+		.window = BT_GAP_SCAN_FAST_WINDOW,
+	};
+
+	UNSET_FLAG(is_connected);
+
+	int err = bt_le_scan_start(&scan_param, device_found);
+
+	ASSERT(!err, "Scanning failed to start (err %d)\n", err);
+
+	LOG_DBG("Central initiating connection...");
+	WAIT_FOR_FLAG(is_connected);
+	LOG_INF("Connected as central");
+
+	/* No security support on the tinyhost unfortunately */
+}
+
+static ssize_t written_to(struct bt_conn *conn,
+			  const struct bt_gatt_attr *attr,
+			  const void *buf,
+			  uint16_t len,
+			  uint16_t offset,
+			  uint8_t flags)
+{
+	static int nwrites;
+
+	LOG_INF("written to: handle 0x%x len %d flags 0x%x",
+		attr->handle,
+		len,
+		flags);
+
+	LOG_HEXDUMP_DBG(buf, len, "Write data");
+
+	if (nwrites == 0) {
+		/* Suspend on the first write, which is an ATT Request */
+		LOG_INF("suspending HCI TX thread");
+		k_thread_suspend(bt_testing_tx_tid_get());
+	}
+
+	if (++nwrites == 3) {
+		SET_FLAG(three_writes);
+	}
```

Comment:
You could consider declaring `nwrites` at global scope as an atomic or volatile value and just increment/decrement/check value as needed; would make it easier to modify the number of writes you expect